main.tf

```diff
@@ -66,7 +67,10 @@ resource "aws_ec2_tag" "cluster_primary_security_group" {
   # This should not affect the name of the cluster primary security group
   # Ref: https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2006
   # Ref: https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2008
-  for_each = { for k, v in merge(var.tags, var.cluster_tags) : k => v if local.create && k != "Name" && var.create_cluster_primary_security_group_tags }
+  # `aws_default_tags` is merged in to "dedupe" tags and stabilize tag updates
+  for_each = { for k, v in merge(var.tags, var.cluster_tags, data.aws_default_tags.current.tags)
+    : k => v if local.create && k != "Name" && var.create_cluster_primary_security_group_tags
```

Comment:
Semicolon char - let's move to previous line

Reply:
I think this is what you were after, let me know if not https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2249/commits/54a35455f672de93c4d21d2667ca3bb3083109c8

Reply:
Yes!